Source/SwiftLintFramework/Rules/Style/SwitchCaseSortRule.swift

```diff
@@ -0,0 +1,243 @@
+                    return nil
+                }
+
+                // sort the multi-item case in itself, then compare the first item with other cases
+                let caseNames = caseLabelSyntax.caseItems.compactMap(sortableName)
+                let caseNamesSorted = caseNames.sorted()
+                if caseNames != caseNamesSorted {
+                    violations.append(caseLabelSyntax.positionAfterSkippingLeadingTrivia)
+                }
+
+                return caseNamesSorted.first
+            }
+
+            let sortedCaseNames = caseNamesToSort.sorted()
+            if caseNamesToSort != sortedCaseNames {
+                violations.append(node.switchKeyword.positionAfterSkippingLeadingTrivia)
+            }
+        }
+    }
+}
+
+private extension SwitchCaseSortRule {
+    final class Rewriter: SyntaxRewriter, ViolationsSyntaxRewriter {
+        private(set) var correctionPositions: [AbsolutePosition] = []
+        private let locationConverter: SourceLocationConverter
+        private let disabledRegions: [SourceRange]
+
+        init(locationConverter: SourceLocationConverter, disabledRegions: [SourceRange]) {
+            self.locationConverter = locationConverter
+            self.disabledRegions = disabledRegions
+        }
+
+        override func visit(_ node: SwitchExprSyntax) -> ExprSyntax {
+            guard node.cases.isNotEmpty,
+                  !node.isContainedIn(regions: disabledRegions, locationConverter: locationConverter)
+            else {
+                return super.visit(node)
+            }
+
+            let newCases = sort(node.cases)
+            if node.cases.description != newCases.description {
+                var newNode = node
+                newNode.cases = newCases
+                correctionPositions.append(node.switchKeyword.positionAfterSkippingLeadingTrivia)
+                return super.visit(newNode)
+            }
+
+            return super.visit(node)
+        }
+    }
+}
+
+private func sort(_ cases: SwitchCaseListSyntax) -> SwitchCaseListSyntax {
+    var elements = [SwitchCaseListSyntax.Element]()
+    for caseListElement in cases {
+        // only sort normal switch cases, not the ones with #if...#endif
+        guard case let .switchCase(caseSyntax) = caseListElement else {
+            return cases
+        }
+        // if it's a case statement with multiple items sort them inside first
+        if case let .case(caseLabelSyntax) = caseSyntax.label,
+           !caseLabelSyntax.caseItems.isEmpty {
+            elements.append(.switchCase(caseSyntax.with(\.label, .case(sort(caseLabelSyntax)))))
+        } else { // default or single cases
+            elements.append(caseListElement)
+        }
+    }
+
+    return SwitchCaseListSyntax(elements.sorted(by: byName))
+}
+
+private func sort(_ caseLabel: SwitchCaseLabelSyntax) -> SwitchCaseLabelSyntax {
+    // remove trailing commas, then sort them, then join them with commas
```

Comment:
The comments are helpful. However, we like to use full sentences ending with a period.